app/components/chart.js

```diff
@@ -128,28 +149,72 @@ export class Chart extends Component {
     const momentRes = getMomentPeriod(resolution);
 
     this.chart.xAxis[0].setExtremes(moment(timestamp).startOf(momentRes).valueOf(), moment(timestamp).endOf(momentRes).valueOf());
+    // this.chart.xAxis[0].update({ tickInterval: 3 * 3600 * 1000 });
     // this.chart.yAxis[0].setExtremes();
     this.chart.redraw();
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
+  componentDidMount() {
+    this.chart = this.refs.chart.getChart();
+    this.draw(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.draw(nextProps);
+  }
+
+  shouldComponentUpdate() {
     return false;
   }
 
   render() {
     return (
-      <ReactHighcharts config={ config } ref="chart" domProps={{ className: 'chart', style: { width: '94%', height: '300px', marginLeft: '3%' } }}></ReactHighcharts>
+      <ReactHighcharts config={ config } ref="chart" domProps={{ className: 'chart', style: { width: '100%', height: '100%' } }}></ReactHighcharts>
     );
   }
 }
 
 function mapStateToProps(state) {
+  const generateRnd = (min, max) => Math.floor(Math.random() * (max - min + 1) + min);
+  const generateValues = () => {
+    switch (state.charts.resolution) {
+      case constants.RESOLUTIONS.YEAR_MONTH:
+        return [...Array(9).keys()].map((i) => ({
+          timestamp: moment().startOf('year').add(i * 1, 'month').valueOf(),
+          value: generateRnd(107284690, 2107284690),
```

Comment:
what are those numbers ? dito with the other resolutions - it just not clear where those numbers come from.